contrib/READMEs/EMISSARY-COMMANDS.md

```diff
@@ -0,0 +1,215 @@
+which is emissary and emissary123.
+
+The default PickUpPlace is configured to read files from _target/data/InputData_.  If you copy
+files into that directory, you will see Emissary process them.  Keep in mind, only toUpper and toLower are
+configured, so the output will be to interesting.
+
+### Agents (Standalone)
+
+The agents command shows you the number of MobileAgents for the configured host and what those
+agents are doing.  By default, the port is 9001, but you can use *-p or --port* to change that.  
+Assuming you are running on 8001 from the server command above, try:
+
+```
+./emissary agents -p 8001
+```
+
+### Pool (Standalone)
+
+Pool is a collapsed view of agents for a node.  It too defaults to port 9001.  To run for the 
+standalone server started above run
+
+```
+./emissary pool -p 8001
+```
+
+This command is more useful for a cluster as it a more digestible view of every node.
+
+### Env
+
+The Env Command requires a server to be running.  It will ask the server for some configuration
+values, like PROJECT_BASE and BIN_DIR.  With no arguments, it will dump an unformatted json response.
+
+```
+./emissary env
+```
+
+But you can also dump a response suitable for sourcing in bash.
+
+```
+./emissary env --bashable
+```
+
+Starting  the Emissary server actually calls this endpoint and dumps out $PROJECT_BASE}/env.sh 
+with the configured variables.  This is done so that shell scripts can `source $PROJECT_BASE}/env.sh` 
+and then have those variable available without having to worry with configuring them elsewhere.
+
+### Run
+
+The Run command is a simple command to execute the main method of the given class.  For example
+
+```
+./emissary run emissary.config.ConfigUtil  <path_to_some_cfg_file>
+```
+
+If you need to pass flags to the main method, use *--* to stop parsing flags and simply pass them along.
+
+```
+./emissary run emissary.config.ExtractResource -- -o outputdir somefile
+```
+
+### Server (Cluster)
+
+Emissary is fun in standalone, but running cluster is more appropriate for real work.  The way to run clustered
+is similar to the standalone, but you need to *-m cluster* to tell the node to connect to other nodes.  In
+clustered mode Emissary will also startup the PickUpClient instead of the PickUpPlace, so you will need to
+start a feeder.
+
+Look at the target/config/peers.cfg to see the rendezvous peers.  In this case, there are 3.  Nodes running
+on port 8001 and 9001 are just Emissary nodes.  The node running on 7001 is the feeder.  So let's start up
+8001 and 9001 in two different terminals.
+
+```
+./emissary server -a 2 -m cluster
+./emissary server -a 2 -m cluster -p 9001
+```
+
+Because these nodes all know about ports 8001, 9001 and 7001, you will see errors in the logs as they
+continue to try to connect.  
+
+Note, in real world deployments we don't run multiple Emissary processes on the same node.  You can configure the
```

Comment:
"Note, it is recommended to only run one Emissary process on any single node." ?

Reply:
There has been some experimentation with running multiple instances on the same node. Maybe we could even remove this guidance? @jpdahlke